cylc/flow/scripts/reinstall.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""cylc reinstall [OPTIONS] ARGS
+
+Reinstall a previously installed workflow.
+
+Examples:
+
+ Having previously installed:
+  $ cylc install myflow
+ To reinstall this workflow run:
+  $ cylc reinstall myflow/run1
+
+ Having previously installed:
+ $ cylc install myflow --no-run-name
+ To reinstall this workflow run:
+  $ cylc reinstall myflow
+
+ To reinstall a workflow from within the cylc-run directory of a previously
+ installed workflow:
+ $ cylc reinstall
```

Comment:
There is some formatting magic in the `--help` output if you write comments like this:

```suggestion
Examples:
  # Having previously installed:
  $ cylc install myflow
  # To reinstall this workflow run:
  $ cylc reinstall myflow/run1

  # Having previously installed:
  $ cylc install myflow --no-run-name
  # To reinstall this workflow run:
  $ cylc reinstall myflow

  # To reinstall a workflow from within the cylc-run directory of a previously
  # installed workflow:
  $ cylc reinstall
```

Reply:
Excellent, good to know, thanks!